beacon-chain/sync/rpc_send_request.go

```diff
@@ -56,12 +57,12 @@ func SendBeaconBlocksByRangeRequest(
 			return nil, ErrInvalidFetchedData
 		}
 		// Returned blocks MUST be in the slot range [start_slot, start_slot + count * step).
-		if blk.Block.Slot < req.StartSlot || blk.Block.Slot >= req.StartSlot+req.Count*req.Step {
+		if blk.Block.Slot < req.StartSlot || blk.Block.Slot >= req.StartSlot.Add(req.Count*req.Step) {
 			return nil, ErrInvalidFetchedData
 		}
 		// Returned blocks, where they exist, MUST be sent in a consecutive order.
 		// Consecutive blocks MUST have values in `step` increments (slots may be skipped in between).
-		if !isFirstChunk && (prevSlot >= blk.Block.Slot || (blk.Block.Slot-prevSlot)%req.Step != 0) {
+		if !isFirstChunk && (prevSlot >= blk.Block.Slot || blk.Block.Slot.SubSlot(prevSlot).Mod(req.Step) != 0) {
```

Comment:
This can panic here, a malicious peer could return a block with a slot less than the previous slot. The go runtime will evaluate both the conditions here leading to a panic due to an underflow.

Reply:
Will make sure I check for `prevSlot <= blk.Block.Slot` before doing any arithmetics on them.